imap_processing/idex/idex_rice_decode.py

```diff
@@ -0,0 +1,205 @@
+# ruff: noqa: PLR0912
+"""
+Decompress IDEX raw wavelength data.
+
+Originally written by Corinne Wuerthner.
+"""
+
+# sub_frame_size is the compression block size
+SUB_FRAME_SIZE = 64
+
+
+# TODO: refactor code blow to use less branches. Remove ruff disabled.
+def _decode_sub_frame(
+    bits: str,
+    bp: int,
+    psel: int,
+    k: int,
+    n_bits: int,
+) -> tuple[list[int], int, int]:
+    """
+    Decode a subframe of compressed data.
+
+    Parameters
+    ----------
+    bits : str
+        Raw waveform binary string.
+    bp : int
+        Current position to start reading from bits.
+    psel : int
+        Predictor select value.
+    k : int
+        Rice parameter used to divide each re-mapped residual value into two parts.
+    n_bits : int
+        Expected number of bits per sample. Either 10 or 12.
+
+    Returns
+    -------
+    tuple[list, int, int]
+        Decompressed subframe as a list of integers, the number of samples and the
+        bit position.
+    """
+    sample_count = 0
+    sub_frame_data = []
+
+    while (sample_count < SUB_FRAME_SIZE) and bp < len(bits):
+        if sample_count == 0:
+            # For every subframe, the first sample is always uncompressed.
+            # Read warmup sample
+            d1, bp = read_bits(bits, bp, n_bits)
+
+            sub_frame_data.append(d1)
+            sample_count += 1
+
+            # A 'psel' value of zero assumes that every sample in the frame is equal
+            # to the first sample in the frame. In this case, the first sample is
+            # stored as the original NBIT binary value
+            if psel == 0:
+                for _ in range(SUB_FRAME_SIZE - 1):
+                    sub_frame_data.append(d1)
```

Comment:
I think this might be equivalent and avoid the for-loop.
```suggestion
                sub_frame_data.extend([d1] * (SUB_FRAME_SIZE - 1))
```